feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,62 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findLowestNoneFreeOffer(validOffers) ?: basePlan
+}
+
+private fun findLongestFreeTrial(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.freePricingPhase()?.let { pricingPhase ->
+            Pair(offer, parseBillPeriodToDays(pricingPhase.billingPeriod))
+        }
+    }.maxByOrNull { it.second }?.first
+}
+
+private fun findLowestNoneFreeOffer(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.nonFreePricingPhase()?.let { pricingPhase ->
+            Pair(offer, pricingPhase.priceAmountMicros)
+        }
+    }.minByOrNull { it.second }?.first
+}
+
+private fun GoogleSubscriptionOption.freePricingPhase(): PricingPhase? {
+    return pricingPhases.firstOrNull()?.takeIf { it.priceAmountMicros == 0L }
+}
+
+private fun GoogleSubscriptionOption.nonFreePricingPhase(): PricingPhase? {
+    return pricingPhases.firstOrNull()?.takeIf { it.priceAmountMicros > 0L }
+}
+
+private const val DAYS_IN_YEAR = 365
+private const val DAYS_IN_MONTH = 30
```

Comment:
did we do any testing on the days in month? like if you start a monthly sub, does it stop 30 days later or 4 weeks or what?

probably doesn't seriously effect the outcome, just curious

Reply:
I pretty certain that a monthly subscription is based on the numerical day so it will generally be 30 or 31 unless February. So I think that 30 is the safest number to assume here and probably shouldn't have any big effect on the outcome 🤷‍♂️